assets/src/edit-story/components/panels/textStyle/font.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { InputGroup, SelectMenu } from '../../form';
+import { MAX_FONT_SIZE, MIN_FONT_SIZE } from '../../../constants';
+import { useFont } from '../../../app/font';
+
+function FontControls({ properties, state, setState }) {
+  const { fontFamily, fontWeight, fontSize } = properties;
+  const {
+    state: { fonts },
+    actions: { getFontWeight, getFontFallback },
+  } = useFont();
+
+  const handleNumberChange = useCallback(
+    (property) => (value) =>
+      setState({ ...state, [property]: parseInt(value) }),
+    [setState, state]
+  );
+  return (
+    <>
+      {fonts && (
+        <SelectMenu
+          label={__('Font family', 'web-stories')}
+          options={fonts}
+          value={state.fontFamily}
+          isMultiple={fontFamily === ''}
+          onChange={(value) => {
+            const currentFontWeights = getFontWeight(value);
+            const currentFontFallback = getFontFallback(value);
+            const fontWeightsArr = currentFontWeights.map(
+              ({ thisValue }) => thisValue
+            );
+            const newFontWeight =
+              fontWeightsArr && fontWeightsArr.includes(state.fontWeight)
+                ? state.fontWeight
+                : 400;
```

Comment:
Right now it doesn't work as expected, to test: change the font from Alef Bold to Alegreya, it should stay on Bold, but it goes back to Normal because `newFontWeight` is always 400/Normal.

`.includes` uses `Object.is` and here we have `Object.is("700", 700) //false`
It's connected to the mentioned `parseInt`. 
We should somehow sync it, from API weight is a string, so either `parseInt` it there or use string all the way.

Reply:
Thanks for catching that! Comes out that `thisValue` was not set and `value` is actually the correct thing to use, that was causing the issue, perhaps something has changed meanwhile from when this logic was first created.

Reply:
Ah, I remembered asking for that value to be renamed once, but thought it hadn't been done.  Good catch, all!